### PR TITLE
Add tests to scheduler that verify backward compatibility of default config

### PR DESCRIPTION
**Pull request by niulechuan**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add a test to capture the working default config from current and previous releases to make sure that continues to load/validate successfully

#### Which issue(s) this PR fixes:
xref: https://github.com/kubernetes/kubernetes/issues/103309
xref: https://github.com/kubernetes/kubernetes/pull/101822#issuecomment-870768062

#### Special notes for your reviewer:
```
None
```
#### Does this PR introduce a user-facing change?
```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```
None
```

### CODE REVIEW

File: cmd/kube-scheduler/app/server_test.go [hunk 1]
```diff
@@ -282,3 +282,420 @@ profiles:
...
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    insecure-skip-tls-verify: true
+    server: %s
+  name: default
+contexts:
+- context:
+    cluster: default
```
Comment (wangyysde) [inline cmd/kube-scheduler/app/server_test.go:312]: how about set this message to a variable, then use it in if statement?

File: cmd/kube-scheduler/app/server_test.go [hunk 2]
```diff
@@ -282,3 +282,420 @@ profiles:
...
+  qps: 50
+enableContentionProfiling: true
+enableProfiling: true
+healthzBindAddress: 0.0.0.0:10251
+kind: KubeSchedulerConfiguration
+leaderElection:
+  leaderElect: true
+  leaseDuration: 15s
+  renewDeadline: 10s
+  resourceLock: leases
```
Comment (wangyysde) [inline cmd/kube-scheduler/app/server_test.go:347]: same as above

File: cmd/kube-scheduler/app/server_test.go (hunk 1 above)
Comment (niulechuan) [reply cmd/kube-scheduler/app/server_test.go]: Thank for the advice, done.

File: cmd/kube-scheduler/app/server_test.go (hunk 2 above)
Comment (niulechuan) [reply cmd/kube-scheduler/app/server_test.go]: @wangyysde Done.

**Review by damemi — COMMENTED** [review]
/ok-to-test

**Pull request closed without merging**